api.go

```diff
@@ -80,6 +80,9 @@ type LineMatch struct {
 	LineEnd    int
 	LineNumber int
 
+	LinesBefore [][]byte
+	LinesAfter  [][]byte
```

Comment:
FYI you now need to include a copy of the byte slices in `copyFiles` function. The underlying file content is mmapped in, so go's garbage collector doesn't see references to it. As such once we have decided on the results to send to the user we call `copyFiles` to ensure the bytes live beyond the search.

Maybe also document that these fields are only set if `NumContextLines` is set.

Reply:
Thanks! I've added them but not sure how to test it properly